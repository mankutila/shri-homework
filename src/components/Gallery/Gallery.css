.gallery {
  display: flex;
  flex-wrap: wrap;
}

.gallery::after {
  content: '';
  flex-grow: 99999;
}