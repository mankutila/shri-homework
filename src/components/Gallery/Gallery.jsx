import React, {Component} from 'react';
import {connect} from 'react-redux';

import {Picture} from "../Picture/Picture";
import {Lightbox} from "../Lightbox/Lightbox";
import {Infinite} from "../Infinite/Infinite";

import {fetchNext} from "../../actions/fetchNext";

import './Gallery.css'


class GalleryComponent extends Component {
  constructor() {
    super();
    this.state = {
      photoIndex: 0,
      isOpen: false
    };
    this.openLightbox = this.openLightbox.bind(this);
  }

  load() {
    const url = 'https://pixabay.com/api/?key=8532246-55268eb0f8f42379b33ae8c5d&q=architecture&image_type=photo&editors_choice=true&safesearch=true&per_page=50';
    fetch(url)
      .then((resp) => resp.json())
      .then((data) => {
        let results = data.hits;
        console.log(results);
        this.props.dispatch({
          type: 'APPEND_IMAGES',
          images: results
        });
      })
      .catch(function (error) {
        console.log(error);
      });
  }

  componentDidMount() {
    this.load();
  }

  openLightbox(index) {
    this.setState({
      isOpen: true,
      photoIndex: index
    });
  }

  render() {
    const {photoIndex, isOpen} = this.state,
          images = this.props.images;
    return (
      <Infinite fetchNext={fetchNext}>
        <section className="gallery">
          {images.map((item, index) => <Picture openLightBox={this.openLightbox}
                                                key={item.id}
                                                index={index}
                                                imgData={item}/>)}
        </section>

        {isOpen && (
          <Lightbox
            src={images[photoIndex].largeImageURL}
            onClose={() => this.setState({isOpen: false})}
            onMovePrev={() =>
              this.setState({
                photoIndex: (photoIndex + images.length - 1) % images.length,
              })
            }
            onMoveNext={() =>
              this.setState({
                photoIndex: (photoIndex + 1) % images.length,
              })
            }
          />
        )}
      </Infinite>
    );
  }
}

const stateToProps = (state) => ({
  images: state.images,
  page: state.page,
  error: state.error
});

export const Gallery = connect(stateToProps)(GalleryComponent);