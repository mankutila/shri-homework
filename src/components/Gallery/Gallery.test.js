import React from 'react';
import { shallow } from 'enzyme';
import { createMockStore } from 'redux-test-utils';
import { Gallery } from './Gallery';

const shallowWithStore = (component, store) => {
  const context = {
    store,
  };
  return shallow(component, { context });
};

it('renders correctly', () => {

  const testState = {
    images: {
      images: [],
      page: 1,
    },
    lightbox: {
      isOpen: false
    },
    tags: {
      current: 'architecture'
    }

  };
  const store = createMockStore(testState);

  const wrapper = shallowWithStore(<Gallery />, store)

  expect(wrapper).toMatchSnapshot();
});
