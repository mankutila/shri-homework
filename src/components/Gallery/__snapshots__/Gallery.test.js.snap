// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<GalleryComponent
  current="architecture"
  dispatch={[Function]}
  images={Array []}
  isOpen={false}
  page={1}
/>
`;
